src/ledmon/ledmon.c

```diff
@@ -971,21 +1036,23 @@ int main(int argc, char *argv[])
 	if (on_exit(_ledmon_fini, progname))
```

Comment:
Regression occurs during testing related to the service.
stopping the ledmon service ends with the failed status, the cause is a segfault that occurs while cleaning up the lists,

`list_next(const struct node * node) (\root\ledmon_open\src\lib\list.h:187)
__list_erase(struct list * list, item_free_t free_fn) (\root\ledmon_open\src\lib\list.c:32)
list_erase(struct list * list) (\root\ledmon_open\src\lib\list.h:94)
sysfs_reset(struct led_ctx * ctx) (\root\ledmon_open\src\lib\sysfs.c:579)
led_free(struct led_ctx * ctx) (\root\ledmon_open\src\lib\libled.c:70)
_ledmon_fini(int status, void * program_name) (\root\ledmon_open\src\ledmon\ledmon.c:173)
libc.so.6!__run_exit_handlers (Unknown Source:0)
libc.so.6!exit (Unknown Source:0)
main(int argc, char ** argv) (\root\ledmon_open\src\ledmon\ledmon.c:1057)`

I looked at the code, and it seems that duplicate cleanup is the cause, because in the line 1056 function `led_free(ctx)` is called, and then it is called again in `_ledmon_fini()` which is triggered in response to exit() call.

It can be reproduced without service:
1. Run ledmon `src/ledmon/ledmon --foreground`
2. Send SIGTERM to ledmon process `kill ledmon_processID`

Reply:
Correction added to PR, thanks